File: src/lib/agentsMd.ts
import { resolveAgentsMd } from "@/lib/tauri";

/**
 * Project-rules loader. Returns the cascading AGENTS.md / CLAUDE.md
 * resolution for `projectPath`, or null when no candidate file exists
 * anywhere in the cascade.
 *
 * The actual walk happens Rust-side in `core::agents_md::resolve` so
 * `~/.claude/AGENTS{,.override}.md` and per-directory candidates from
 * git-root down to cwd are concatenated atomically with the 32 KiB cap
 * applied to the joined output (matches Codex CLI 0.122+'s behavior).
 *
 * Failures are swallowed — a missing project / unreadable file is the
 * common case, never blocking a launch.
 */
export async function loadAgentsMd(
  projectPath: string,
): Promise<string | null> {
  if (!projectPath) return null;
  try {
    return await resolveAgentsMd(projectPath);
  } catch {
    return null;
  }
}

File: src-tauri/src/commands/agents_md.rs
//! A4 — Tauri command surface for the AGENTS.md cascading resolver.
//!
//! Returns the concatenated cascade body (or `null` when nothing is found)
//! so the frontend's `loadAgentsMd` helper stays a single round-trip
//! instead of one read per candidate file.

use crate::core::agents_md;

/// Resolve the AGENTS.md / CLAUDE.md cascade for `cwd`. See
/// `core::agents_md::resolve` for precedence rules.
#[tauri::command]
pub async fn resolve_agents_md(cwd: String) -> Result<Option<String>, String> {
    super::validate_project_path(&cwd)?;
    Ok(agents_md::resolve(&cwd))
}
